syntax = "proto2";

message Config {
    message Tesla {
        string user_id = 1;
        // Location of the cache json file generated by TeslaPy for authentication
        string cache_file = 2 [default = '/app/config/cache.json'];
    }

    message IAquaLink {
        message Pool {
            // By decreasing order of priority (i.e. devices should be turned on in this order and turned off in the reverse order)
            repeated string devices = 1;
            // Minimum excess power required to activate a pool device (in Watt)
            int activation_excess_power = 2 [default = 2000];
            // Excess power threshold below which pool devices start to be deactivated (in Watt)
            int deactivation_excess_power = 3 [default = 500];
            // Pool will not turn on before 8h00
            int operating_window_start = 4 [default = 8];
            // Pool will be shut off in any case after 18h00
            int operating_window_end = 5 [default = 18];
        }

        message Light {
            repeated string devices = 1;
            // Minimum battery required to turn on the lights, in %
            int minimum_battery = 2 [default = 50];
            // Lights will be shut off in any case after 22h00
            int operating_window_end = 3 [default = 22];
        }

        string user_id = 1;
        string password = 2;
        Pool pool = 3;
        Light light = 4;
    }

    message Location {
        // pytz time zone
        string timezone = 1 [default = 'UTC'];
        // Northern Latitude
        float latitude = 2;
        // Eastern Longitude
        float longitude = 3;
    }

    Tesla tesla = 1;
    IAquaLink iaqualink = 2;
    Location location = 3;
}
